app/parse.py

```diff
@@ -15,8 +24,120 @@ class Product:
     num_of_reviews: int
 
 
+PRODUCT_FIELDS = [field.name for field in fields(Product)]
+
+
+def write_products_to_csv(products: list[Product], csv_file: str) -> None:
+    with open(csv_file, "w", encoding="utf-8", newline="") as output_csv_file:
+        writer = csv.writer(output_csv_file)
+        writer.writerow(PRODUCT_FIELDS)
+        writer.writerows([astuple(product) for product in products])
+
+
+def get_driver() -> webdriver:
+    chrome_options = Options()
+    chrome_options.add_argument("--headless")
+    chrome_options.add_argument("--window-size=%s" % WINDOW_SIZE)
+    driver = webdriver.Chrome(options=chrome_options)
+
+    return driver
+
+
+def parse_hdd_block_prices(product_element: WebElement) -> dict[str, float]:
+    detailed_url = urljoin(
+        BASE_URL,
+        product_element.find_element(By.CLASS_NAME, "title").get_attribute(
+            "href"
+        ),
+    )
+    driver = get_driver()
+    driver.get(detailed_url)
+    try:
+        swatches = driver.find_element(By.CLASS_NAME, "swatches")
+        buttons = swatches.find_elements(By.TAG_NAME, "button")
+
+        prices = {}
+
+        for button in buttons:
+            if not button.get_property("disabled"):
+                button.click()
+                prices[button.get_property("value")] = float(
+                    driver.find_element(By.CLASS_NAME, "price").text.replace(
+                        "$", ""
+                    )
+                )
+        driver.close()
+        return prices
+    except NoSuchElementException:
+        pass
```

Comment:
add more informative way if exception occurs
use print() function instead of pass